src/fping.c

```diff
@@ -2845,33 +3101,57 @@ int wait_for_reply(int64_t wait_time)
         }
 
         if (icmp_request_typ == 13) {
-            printf("%s timestamps: Originate=%u Receive=%u Transmit=%u Localreceive=%u",
-                   alive_flag ? "" : ",",
-                   ip_header_otime_ms, ip_header_rtime_ms, ip_header_ttime_ms,
-                   ms_since_midnight_utc(recv_time));
+            if (json_flag) {
+              printf(", \"timestamps\": {");
+              printf("\"originate\": %u, ", ip_header_otime_ms);
+              printf("\"receive\": %u, ", ip_header_rtime_ms);
+              printf("\"transmit\": %u, ", ip_header_ttime_ms);
+              printf("\"localreceive\": %u}", ms_since_midnight_utc(recv_time));
+            }
+            else {
+                printf("%s timestamps: Originate=%u Receive=%u Transmit=%u Localreceive=%u",
+                      alive_flag ? "" : ",",
+                      ip_header_otime_ms, ip_header_rtime_ms, ip_header_ttime_ms,
+                      ms_since_midnight_utc(recv_time));
+            }
         }
 
         if(print_tos_flag) {
             if(ip_header_tos != -1) {
-                printf(" (TOS %d)", ip_header_tos);
+                if (json_flag)
+                    printf(", \"tos\": %d", ip_header_tos);
+                else
+                    printf(" (TOS %d)", ip_header_tos);
             }
             else {
-                printf(" (TOS unknown)");
+                if (json_flag)
+                    printf(", \"tos\": -1");
+                else
+                    printf(" (TOS unknown)");
             }
         }
 
         if (print_ttl_flag) {
           if(ip_header_ttl != -1) {
-              printf(" (TTL %d)", ip_header_ttl);
+              if (json_flag)
+                  printf(", \"ttl\": %d", ip_header_ttl);
+              else
+                  printf(" (TTL %d)", ip_header_ttl);
           }
           else {
-              printf(" (TTL unknown)");
+              if (json_flag)
+                  printf(", \"ttl\": -1");
```

Comment:
I would prefer the same output value for JSON and normal output, i.e. `"unknown"` instead of `-1`.